lib/active_decorator/monkey/active_record/associations.rb

```diff
@@ -22,6 +22,24 @@ def first_nth_or_last(*)
             end
           end
         end
+
+        module CollectionProxy
+          def find_nth(index)
+            ActiveDecorator::Decorator.instance.decorate_association(@association.owner, super)
+          end
+
+          def find_nth_with_limit(index, limit)
+            ActiveDecorator::Decorator.instance.decorate_association(@association.owner, super)
+          end
+
+          def last(limit = nil)
+            ActiveDecorator::Decorator.instance.decorate_association(@association.owner, super)
+          end
+
+          def find_nth_from_last(index)
+            ActiveDecorator::Decorator.instance.decorate_association(@association.owner, super)
+          end
+        end
```

Comment:
I added support for `take` in #77 , so please merge #77 first. I'll rebase and fix tests after merging.